homeassistant/components/atome/sensor.py

```diff
@@ -0,0 +1,319 @@
+        return self._monthly_price
+
+    def get_monthly_usage(self):
+        """Return current monthly power usage."""
+        try:
+            values = self.atome_client.get_consumption(MONTHLY_TYPE)
+            self._monthly_usage = values["total"] / 1000
+            self._monthly_price = values["price"]
+            _LOGGER.debug("Updating Atome monthly data. Got: %d.", self._monthly_usage)
+
+        except KeyError as error:
+            _LOGGER.error("Missing last value in values: %s: %s", values, error)
+            return None
+
+    @property
+    def yearly_usage(self):
+        """Return latest yearly usage value."""
+        return self._yearly_usage
+
+    @property
+    def yearly_price(self):
+        """Return latest yearly usage value."""
+        return self._yearly_price
+
+    def get_yearly_usage(self):
+        """Return current yearly power usage."""
+        try:
+            values = self.atome_client.get_consumption(YEARLY_TYPE)
+            self._yearly_usage = values["total"] / 1000
+            self._yearly_price = values["price"]
+            _LOGGER.debug("Updating Atome yearly data. Got: %d.", self._yearly_usage)
+
+        except KeyError as error:
+            _LOGGER.error("Missing last value in values: %s: %s", values, error)
+            return None
+
+
+class AtomeSensor(Entity):
+    """Representation of a sensor entity for Atome."""
+
+    def __init__(self, data, name, sensor_type, update_call):
+        """Initialize the sensor."""
+        self._name = name
+        self._data = data
+        self._state = None
+        self._attributes = {}
+
+        self._sensor_type = sensor_type
+        self.update_sensor = update_call
+
+        if sensor_type == LIVE_TYPE:
+            self._unit_of_measurement = POWER_WATT
+        else:
+            self._unit_of_measurement = ENERGY_KILO_WATT_HOUR
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name or DEFAULT_NAME
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes."""
+        return self._attributes
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return ICON
+
+    @property
+    def device_class(self):
+        """Return the device class."""
+        return DEVICE_CLASS_POWER
+
+    def update(self):
+        """Update device state."""
+        self.update_sensor()
+
+        if self._sensor_type == LIVE_TYPE:
+            self._state = self._data.live_power
+            self._attributes["subscribed_power"] = self._data.subscribed_power
+            self._attributes["is_connected"] = self._data.is_connected
```

Comment:
Is `is_connected` different from `available` property?

Reply:
In my opinion it is:
the Atome device sends regularly the energy data to some IOT API.
here, we read this API. The API returns the fact that the Atome device is physically connected or not, but even though the device is disconnected, the API remains available.

Reply:
Ok.